### ISSUE TITLE
Issues with Lightgun Configuration with Naomi

### ISSUE THREAD

**Issue by mantis-droid** [libretro/flycast#873]
Hello - I wanted to note that it appears I'm having an issue using the latest build of the Flycast core during lightgun configuration of the majority of Naomi games.

Specifically with Maze of Kings within the test menu, using an Aimtrak it is recording trigger presses in calibration but I'm getting "00H" on all areas of the screen. Similar behavior seems to be taking place with HOTD2, Ninja Assault, etc. For reference though, I haven't encountered any similar issues with other emulators (MAME, TP, etc.) and it seems to be tied to Naomi/Flycast. This sounds similar to what was being reported in #537 where that was a "80H" error but that seems to have been resolved.

Has anyone encountered the issue previously? I'm wondering if there was a method for resolving this if it has been previously posted. Thanks for the help

**Comment by flyinghead** [comment]
Make sure to set your controller type as Light Gun. You can also enable the gun crosshair in the core options to check if your control is being recognized by flycast. I just tested Mazan and ninja assault without issue.

**Comment by mantis-droid** [comment]
Hi flyinghead - Thanks for the quick response. I should provide some additional information as to how I've been setting up my config. Again this was using two separate Aimtraks:

- Load up Maze of Kings using latest RA/Flycast
- Change config to Controller to be able to enter Service mode (Mouse works fine as well). No buttons are recognized to enter Service mode as Light Gun
- Navigate to Gun Calibration as Controller as Controls
- Switch to Light Gun in Controls in RA - ROM detects both Aimtraks, crosshairs detected. Registers "Sight Out" in controls when taking crosshairs off screen as expected
- Follow through config process as expect, ROM detects trigger presses, registers as "00H" on all areas of the screen

Should also note that both guns seem to work as intended within Aimtrak utility, not sure I mentioned that earlier.

Its interesting that you're getting different results, are you taking a similar approach to your config?

**Comment by flyinghead** [comment]
That's pretty much exactly what I'm doing (and having to switch back and forth between Gamepad and Lightgun when using the service menu...)
Make sure to navigate out of the service menu so that the settings are saved to the eeprom file. If your control is working fine during calibration, it should be fine in game.

**Comment by flyinghead** [comment]
One more thing: Maze of the Kings is not registered as a game using a lightgun (which it should) so make sure to start the game with at least one controller as lightgun. It may make a difference. I'll fix this.

**Comment by mantis-droid** [comment]
Yep typically what I'll do is that once the gun configurations complete, I'll go back into RA to switch the controls back to Controller and go through the standard save process. Then once the game goes and completes boot I'll exit out of the game in RA, change controls from Controller to Light Gun and start back up.

Its frustrating because on all the light guns games I've tried (must be close to all of them on the Compatibility List marked as capable at this point), I can start the game/see the crosshairs/hear the fire effect but not hits are being registered on the screen. I feel like I'm so close on this lol

But thanks for taking a look at this for me, let me know if theres any additional info I can grab for you that would help

**Comment by flyinghead** [comment]
I have fixed some issues related to lightgun. Please re-test with the latest build.

**Comment by mantis-droid** [comment]
Hello - Sorry about the delay in response, no change unfortunately. Although for reference, the Ninja Assault issue noted elsewhere I am having as well so I can at least confirm that part works
